# HG changeset patch
# User Carlo de Falco <carlo.defalco@polimi.it>
# Date 1478064504 -3600
#      Wed Nov 02 06:28:24 2016 +0100
# Node ID 51c9eded34ee995a3d40963130d6e1208dd2461f
# Parent  284bbb0328f2edf58a2425e13daee49151d17c57
Fix access method to sundials 2.7.0 sparse matrix structure.

* libinterp/dldfunctions/__ode15__.cc : dereference twice colptrs/rowvals to
  access sundials sparse matrix structures.

diff --git a/libinterp/dldfcn/__ode15__.cc b/libinterp/dldfcn/__ode15__.cc
--- a/libinterp/dldfcn/__ode15__.cc
+++ b/libinterp/dldfcn/__ode15__.cc
@@ -392,23 +392,25 @@ namespace octave
     SparseMatrix jac;
 
     if (havejacfun)
       jac = (*jacspfun) (y, yp, t, cj, ida_jac);
     else
       jac = (*jacspcell) (spdfdy, spdfdyp, cj);
 
     SparseSetMatToZero (Jac);
-
+    int *colptrs = *(Jac -> colptrs);
+    int *rowvals = *(Jac -> rowvals);
+    
     for (int i = 0; i < num + 1; i++)
-      *(Jac -> colptrs)[i] = jac.cidx(i);
+      colptrs[i] = jac.cidx(i);
 
     for (int i = 0; i < jac.nnz (); i++)
       {
-        *(Jac -> rowvals)[i] = jac.ridx (i);
+        rowvals[i] = jac.ridx (i);
         Jac -> data[i] = jac.data (i);
       }    
 
     END_INTERRUPT_WITH_EXCEPTIONS;
   }
 
   ColumnVector
   IDA::NVecToCol (N_Vector& v, long int n)
